website/docs/plugin/code-generation/specification.mdx

```diff
@@ -0,0 +1,692 @@
+
+## attribute_types
+
+An object within the `attribute_types` array is defined as the following.
+
+Note that `attribute_types` can **only** be defined on `object` attributes.
+
+One of `bool`, `float64`, `int64`, `list`, `map`, `number`, `object`, `set` or `string` **must** be specified.
+
+```json
+{
+  "datasources": [
+    {
+      "name": "example",
+      "schema": {
+        "attributes": [
+          {
+            "name": "object_attribute",
+            "object": {
+              "computed_optional_required": "optional",
+              "attribute_types": [
+                {
+                  "name": "example",
+                  "string": {}
+                }
+              ]
+            }
+          },
+          ...
+        ]
+      }
+    }
+  ],
+  ...
+}
+```
+
+| Property      | Type   | Required | Description                                           |
+|---------------|--------|----------|-------------------------------------------------------|
+| bool          | Object | No       | Boolean [schema type](#schema-type) definition.       |
+| float64       | Object | No       | Float64 [schema type](#schema-type) definition.       |
+| int64         | Object | No       | Int64 [schema type](#schema-type) definition.         |
+| list          | Object | No       | List [schema type](#schema-type) definition.          |
+| list_nested   | Object | No       | List nested [schema type](#schema-type) definition.   |
+| map           | Object | No       | Map [schema type](#schema-type) definition.           |
+| map_nested    | Object | No       | Map nested [schema type](#schema-type) definition.    |
+| name          | String | Yes      | The name of the attribute type.                       |
+| number        | Object | No       | Number [schema type](#schema-type) definition.        |
+| object        | Object | No       | Object [schema type](#schema-type) definition.        |
+| set           | Object | No       | Set [schema type](#schema-type) definition.           |
+| set_nested    | Object | No       | Set nested [schema type](#schema-type) definition.    |
+| single_nested | Object | No       | Single nested [schema type](#schema-type) definition. |
+| string        | Object | No       | String [schema type](#schema-type) definition.        |
+
+## associated_external_type
+
+The `associated_external_type` object is defined as the following. It provides a mechanism for defining the relationship between a Terraform Plugin Framework type, and another Go type. This is useful during code generation as it enables the auto-generation of methods which convert to the external type from the Framework type and vice versa.
+
+Note that `associated_external_type` can **only** be specified on a [nested_object](#nested_object) within `list_nested`, `map_nested`, and `set_nested` attributes, or on a `single_nested` attribute.
```

Comment:
This is currently in flux. It is true for the initial release but will soon change as we include primitive attributes, collections etc. Not sure whether we should leave this as is or re-word to indicate upcoming changes?

Reply:
talked about this OOB and we'll keep it as-is and come back and adjust later 👍🏻